src/translators/sql_translator.py

```diff
@@ -153,16 +199,21 @@ def _insert_entities_of_type(self,
                 msg = "Translating entity without TIME_INDEX. " \
                       "It should have been inserted by the 'Reporter'. {}"
                 warnings.warn(msg.format(e))
-                now_iso = datetime.utcnow().isoformat(timespec='milliseconds')
-                e[self.TIME_INDEX_NAME] = now_iso
+                e[self.TIME_INDEX_NAME] = current_timex()
+
+            if ORIGINAL_ENTITY_COL in e:
+                raise ValueError(
+                    f"Entity {e[NGSI_ID]} has a reserved attribute name: " +
+                    "'{ORIGINAL_ENTITY_COL_NAME}'")
 
         # Define column types
         # {column_name -> crate_column_type}
         table = {
             'entity_id': self.NGSI_TO_SQL['Text'],
             'entity_type': self.NGSI_TO_SQL['Text'],
             self.TIME_INDEX_NAME: self.NGSI_TO_SQL[TIME_INDEX],
-            FIWARE_SERVICEPATH: self.NGSI_TO_SQL['Text']
+            FIWARE_SERVICEPATH: self.NGSI_TO_SQL['Text'],
+            ORIGINAL_ENTITY_COL: self.NGSI_TO_SQL[NGSI_TEXT]
```

Comment:
why not NGSI_STRUCTURED_VALUE?

Reply:
yea, that's what i did initially, but then i thought plain text would be more bomb proof. in fact, if we use a db json field, the db parser may not necessarily agree w/ the python serializer and puke at us. so plain text is the safest---see my comment above about it.

Reply:
i thought about "object" so that we could query it and eventually use it to "fix" data

Reply:
stashing away the original entity as json would have some advantages though, e.g. we can run json queries on that column to debug failures---postgres has an integrated json query lang, not sure about crate. perhaps, we should store it as json, since the likelihood of python producing a json the sql engine won't be able to parse is, frankly, just a corner case in my opinion---famous last words.

Reply:
> i thought about "object" so that we could query it and eventually use it to "fix" data

yep, see my comment, i think you commented just before i hit the comment button :-)
should we do json then? which is more important to you bomb proof insert or being able to query the json?

Reply:
if think we store as a json, and ifthe message is not "json", we log the error